### PR TITLE
head_to_head method and test

### PR DESCRIPTION
**Pull request by sethverrill**
I'm not exactly sure why, but I couldn't get this to work without putting the team name method into this class.
Even when I was calling it from stat_tracker, I could get results in pry, but not in the method itself.

### CODE REVIEW

**Review by Jonathan-Atkins — APPROVED** [review]
nice work

**Comment by MiTOBrien** [comment]
Approving this, but I'm not sure it is correct?  Maybe I'm overlooking it in the code, but I don't see where you're getting the win percentage for the hash?
The specification is:

Record (as a hash - win/loss) against all opponents with the opponents’ names as keys and the win percentage against that opponent as a value.

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -44,4 +44,39 @@ def fewest_goals_scored(team_id)
...
+                 puts "Game #{game.game_id}: #{team_name(team_id)} lost against #{team_name(opp_team_id)}"
+            end
+          end
+        end
+        
+        results.transform_keys! { |id| team_name(id) }
+        results
+    end
+
+    def team_name(team_id)
```
Comment (Jonathan-Atkins) [inline lib/team_statistics.rb:77]: do we now have two methods for team_name in this file?

**Review by Jonathan-Atkins — COMMENTED** [review]
do we now have two methods for team_name?

**Pull request merged**